querybook/server/datasources/metastore.py

```diff
@@ -98,6 +98,38 @@ def get_table_by_name(
     return table_dict
 
 
+@register("/table_name/<schema_name>/<table_name>/exists/", methods=["GET"])
+def get_if_schema_or_table_exists(
+    metastore_id, schema_name, table_name
+) -> Tuple[bool, bool]:
+    """
+    Check if the table name / schema name exists in cache, then check the actual metastore
+    if they don't exist
```

Comment:
is it possible that it exists in cache while it has been deleted in the actual metastore?

Reply:
yeah it is, I think this is more like a conservative check. My concern with directly with metastore is that we don't have efficient API to do so